pkgs/tools/misc/empty/default.nix

```diff
@@ -0,0 +1,49 @@
+{ fetchgit, lib, stdenv, which }:
+
+let
+  baseName = "empty";
+  rev = "6e81f477dba6c2f9086da70b02abec71b64fb16e";
+  baseVersion = "0.6.21b";
+  version = baseVersion + "+1-git-" + (builtins.substring 0 7 rev);
+  name = baseName + "-" + version;
+in stdenv.mkDerivation {
+  inherit name;
+
+  src = fetchgit {
+    inherit rev;
+    url = "https://git.code.sf.net/p/${name}/code";
+    sha256 = "06mb6vvc8snn3c3zzk0qckz5dpdqh5p1485sh7q1wpj8rvbws9dd";
+  };
+
+  nativeBuildInputs = [ which ];
+
+  makeFlags = [ "PREFIX=$(out)" ];
+
+  # Allow the installation previx to be overridden, and fix the CC parameter
+  # order.
+  patches = [ ./allow-prefix-override.patch ./fix-cc-parameter-order.patch ];
+
+  meta = with lib; {
+    homepage = "https://empty.sourceforge.net";
+    description =
+      "A program to run processes and applications under pseudo-terminal (PTY) sessions and replace TCL/Expect with a simple shell-tool";
+    license = licenses.bsd3;
+    platforms = platforms.all;
+    longDescription = ''
+      The empty utility provides an interface to execute and/or interact with
+      processes under pseudo-terminal sessions (PTYs). This tool is definitely
+      useful in programming of shell scripts designed to communicate with
+      interactive programs like telnet, ssh, ftp, etc. In some cases empty can
+      be the simplest replacement for TCL/expect or other similar programming
+      tools because empty:
+
+      - can be easily invoked directly from shell prompt or script
+      - does not use TCL, Perl, PHP, Python or anything else as an underlying language
+      - is written entirely in C
+      - has small and simple source code
+      - can easily be ported to almost all UNIX-like systems
+    '';
+    maintainers = [ maintainers.djwf ];
+    priority = 10;
```

Comment:
Can you add a comment why we want this?

Reply:
So this program also contains the expect binary which conflicts with expect?

Reply:
Nope: it contains the binary "empty" that performs tasks similar to expect, but using a different interface. There's a really good example on the home page (http://empty.sourceforge.net) that shows examples using empty, TCL/Expect, Perl/expect module, and Python/Pexpect.

Reply:
```suggestion
```

Then I don't think we need this.